packages/components/src/components/lifecycleValidation.spec.ts

```diff
@@ -136,6 +148,20 @@ it.each<TagName>(tagNamesPublicWithoutProps)('should not call validateProps() fo
   expect(spy).not.toBeCalled();
 });
 
+it.each<TagName>(tagNamesWithPropsOfTypeObject)(
+  'should call isDeepEqual() with correct parameters for %s',
+  (tagName) => {
+    const spy = jest.spyOn(isDeepEqualUtils, 'isDeepEqual');
+    const component = componentFactory(tagName);
+
+    if (component.componentShouldUpdate) {
+      component.componentShouldUpdate('newValue', 'oldValue');
+    }
+
+    expect(spy).toBeCalledWith('newValue', 'oldValue');
```

Comment:
The part where the return value of `componentShouldUpdate()` should be the opposite of what `isDeepEqual()` returns is missing?